### PR TITLE
Undo upvote

### PR DESCRIPTION
**Pull request by sbrych**

### CODE REVIEW

File: app/react/components/props/vote-test.jsx [hunk 1]
```diff
@@ -69,4 +71,34 @@ describe('props/vote', () => {
...
+      it('is visible', () => {
+        expect(() => {
+          TestUtils.findRenderedDOMComponentWithClass(componentWithUndoButton, 'undo-upvote-button');
+        }).toNotThrow;
+      });
+
+      it('runs passed undoUpvote handler when clicked', () => {
+        const element = TestUtils
+          .findRenderedDOMComponentWithClass(componentWithUndoButton, 'undo-upvote-button');
+        console.log(element);
```
Comment (krzkol) [inline app/react/components/props/vote-test.jsx]: Is it necessary here?

File: spec/requests/api/v1/props_spec.rb [hunk 2]
```diff
@@ -117,4 +117,42 @@
...
+      it 'returns unathorized response' do
+        delete "/api/v1/props/#{prop.id}/undo_upvotes"
+        expect(response).to have_http_status(401)
+      end
+    end
+
+    context 'user tries to undo upvote of different user' do
+      before do
+        sign_in(user2)
+      end
```
Comment (krzkol) [inline spec/requests/api/v1/props_spec.rb]: It can be surrounded with {} like after block.

File: spec/requests/api/v1/props_spec.rb [hunk 3]
```diff
@@ -117,4 +117,42 @@
...
+
+      it 'undoes the upvote' do
+        delete "/api/v1/props/#{prop.id}/undo_upvotes"
+        expect(json_response['errors']).to eq(I18n.t('props.errors.no_upvote'))
+      end
+    end
+    context 'user undoes own upvote' do
+      before do
+        sign_in(user)
+      end
```
Comment (krzkol) [inline spec/requests/api/v1/props_spec.rb]: Similar as above.

File: app/react/components/props/vote-test.jsx (hunk 1 above)
Comment (sbrych) [inline app/react/components/props/vote-test.jsx]: Good catch :]

**Comment by krzkol** [comment]
Looks good 👍

**Pull request merged**